File: taskmanager/src/main/java/com/tanjer/taskmanager/service/TaskService.java
package com.tanjer.taskmanager.service;

import com.tanjer.taskmanager.entity.Task;
import com.tanjer.taskmanager.exception.ResourceNotFoundException;
import com.tanjer.taskmanager.repository.TaskRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class TaskService {

    @Autowired
    private TaskRepository taskRepository;

    public Task saveTask(Task task) {
        return taskRepository.save(task);
    }

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Task getTaskById(Long id) {
        return taskRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Task not found with id: " + id));
    }

    public Task updateTask(Long id, Task taskDetails) {
        Task existingTask = getTaskById(id);
        
        existingTask.setTitle(taskDetails.getTitle());
        existingTask.setDescription(taskDetails.getDescription());
        existingTask.setStatus(taskDetails.getStatus());
        existingTask.setPriority(taskDetails.getPriority());
        
        return taskRepository.save(existingTask);
    }

    public void deleteTask(Long id) {
        // Ensure the task exists before deleting; will throw exception if not found
        Task existingTask = getTaskById(id);
        taskRepository.delete(existingTask);
    }
}

File: taskmanager-frontend/src/services/TaskService.js
import axios from 'axios';
const API_BASE_URL = import.meta.env.VITE_API_URL ? `${import.meta.env.VITE_API_URL}/tasks` : "https://taskmanager-backend-wct1.onrender.com/tasks";
const TaskService = {
  getAllTasks: () => {
    return axios.get(API_BASE_URL);
  },
  getTaskById: (id) => {
    return axios.get(`${API_BASE_URL}/${id}`);
  },
  createTask: (task) => {
    return axios.post(API_BASE_URL, task);
  },
  updateTask: (id, taskDetails) => {
    return axios.put(`${API_BASE_URL}/${id}`, taskDetails);
  },
  deleteTask: (id) => {
    return axios.delete(`${API_BASE_URL}/${id}`);
  }
};
export default TaskService;

